challenge.rb

```diff
@@ -1,26 +1,28 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.map &:capitalize
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.select { |animal| animal == "dog" }
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.reject { |animal| animal == "dog" }
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.length
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  input.take 2
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  # alternatively: input.select { |animal| animal =~ /cat|dog/ }, optionally with the regex 'i' option
+  # but I already used that and intersection is more concise.
+  input & ["cat", "dog"]
```

Comment:
Oh! That's what that meant, I will be amending that using `keep_if`. Thanks! :+1: